### ISSUE TITLE
With `open-external-link-in-new-window-p` set to `t` and internal buffer as current buffer, `set-url` opens a new window

### ISSUE THREAD

**Issue by edgar-vincent** [atlas-engineer/nyxt#1140]
**Describe the bug**

**Precise recipe to reproduce the issue**

With `open-external-link-in-new-window-p` set to `t`, opening a URL with `set-url` with an internal buffer as current buffer will open a new window, although it is not an external link.

**Information**
- OS name+version: Arch Linux
- Graphics card and driver: Intel Inegrated Graphics
- Desktop environment / Window manager name+version: Gnome 3.38.3
- How you installed Nyxt (Guix pack, package manager, build from source): nyxt-browser-git from the AUR
- Information from command copy-system-information:
```
Nyxt version: 2-pre-release-5-438-g5355f34e 
Operating system kernel: Linux 5.10.13-arch1-2
Lisp implementation: SBCL 2.1.1
Features: (WEBKIT2 WEBKIT2-2.30 WEBKIT2-TRACKING WEBKIT2-MUTE WEBKIT2-EMOJI
           WEBKIT2-MEDIA WEBKIT2-SANDBOXING GTK-3-22 GTK-3-20 GTK-3-18 GTK-3-16
           GTK-3-14 GTK-3-12 GTK-3-10 GTK-3-8 GTK-3-6 GTK-3-4 GTK GDK-3-22
           GDK-3-20 GDK-3-18 GDK-3-16 GDK-3-14 GDK-3-12 GDK-3-10 GDK-3-8
           GDK-3-6 GDK-3-4 CAIRO-1-10 CAIRO-1-12 GDK-PIXBUF GLIB-2-30 GLIB-2-32
           GLIB-2-34 GLIB-2-36 GLIB-2-38 GLIB-2-40 GLIB-2-42 GLIB-2-44
           GLIB-2-46 GLIB-2-48 GLIB-2-50 GLIB-2-52 GLIB-2-54 GLIB-2-56
           GLIB-2-58 GLIB FSET-EXT-STRINGS CUSTOM-HASH-TABLE-NATIVE SWANK
           PLUMP-UTF-32 GLOBAL-VARS DECLARE-TYPES PARENSCRIPT NAMED-READTABLES
           OSICAT-FD-STREAMS 21BIT-CHARS CHUNGA FLEXI-STREAMS CLOSER-MOP
           CL-PPCRE-UNICODE CL-UNICODE CL-PPCRE BORDEAUX-THREADS SPLIT-SEQUENCE
           GRAY-STREAMS FLAT-NAMESPACE X86-64 UNIX CFFI FLAT-NAMESPACE
           SEQUENCE-EMPTYP FAST-IO-SV FAST-IO SBCL-USES-SB-ROTATE-BYTE
           ASDF-SYSTEM-CONNECTIONS CL-JSON-CLOS CL-JSON THREAD-SUPPORT
           QUICKLISP ASDF3.3 ASDF3.2 ASDF3.1 ASDF3 ASDF2 ASDF OS-UNIX
           NON-BASE-CHARS-EXIST-P ASDF-UNICODE X86-64 GENCGC 64-BIT ANSI-CL
           COMMON-LISP ELF IEEE-FLOATING-POINT LINUX LITTLE-ENDIAN
           PACKAGE-LOCAL-NICKNAMES SB-CORE-COMPRESSION SB-LDB SB-PACKAGE-LOCKS
           SB-THREAD SB-UNICODE SBCL UNIX)
```

**Output when started from a shell**

With `init.lisp` containing `(define-configuration browser ((open-external-link-in-new-window-p :t)))`:
```
[user@host ~]$ nyxt --verbose
Arguments parsed: (VERBOSE T URLS NIL) and NIL
 <INFO> [19:05:44] nyxt start.lisp (listen-or-query-socket) -
  Listening to socket "/home/user/.local/share/nyxt/nyxt.socket".
Nyxt version 2-pre-release-5-438-g5355f34e
 <INFO> [19:05:44] nyxt start.lisp (load-lisp unsafe-load) -
  Loading Lisp file "/home/user/.config/nyxt/init.lisp".
<DEBUG> [19:05:44] nyxt renderer-gtk.lisp (ffi-initialize gtk-browser) -
  Initializing GTK Interface
<DEBUG> [19:05:44] nyxt base-mode.lisp (base-mode form-fun-26) -
  BASE-MODE enabled.
<DEBUG> [19:05:44] nyxt/web-mode web-mode.lisp (web-mode form-fun-30) -
  WEB-MODE enabled.
<DEBUG> [19:05:44] nyxt/help-mode help-mode.lisp (help-mode form-fun-24) -
  HELP-MODE enabled.
 <INFO> [19:05:45] nyxt message.lisp (echo form-fun-6) -
  Loading history of xxx URLs from "/home/user/.local/share/nyxt/history/default.lisp".
<DEBUG> [19:05:45] nyxt/minibuffer-mode minibuffer-mode.lisp (minibuffer-mode form-fun-26) -
  MINIBUFFER-MODE enabled.
<DEBUG> [19:05:47] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "down" NYXT::KEYCODE: 116 CHARACTER: #\Nul
  NYXT::KEYVAL-NAME: "Down" 
<DEBUG> [19:05:47] nyxt input.lisp (dispatch-input-event) -
  Found key binding "down" [#116] to SELECT-NEXT
<DEBUG> [19:05:50] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "return" NYXT::KEYCODE: 36 CHARACTER: #\Return
  NYXT::KEYVAL-NAME: "Return" 
<DEBUG> [19:05:50] nyxt input.lisp (dispatch-input-event) -
  Found key binding "return" [#36] to RETURN-SELECTION
<DEBUG> [19:05:54] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: NIL NYXT::KEYCODE: 37 CHARACTER: #\Nul
  NYXT::KEYVAL-NAME: "Control_L" 
<DEBUG> [19:05:56] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "l" NYXT::KEYCODE: 46 CHARACTER: #\l NYXT::KEYVAL-NAME: "l"
  NYXT::MODIFIERS: ("control") 
<DEBUG> [19:05:56] nyxt input.lisp (dispatch-input-event) -
  Found key binding "C-l" [C-#46] to SET-URL-FROM-CURRENT-URL
<DEBUG> [19:05:56] nyxt/minibuffer-mode minibuffer-mode.lisp (minibuffer-mode form-fun-26) -
  MINIBUFFER-MODE enabled.
<DEBUG> [19:05:56] nyxt/minibuffer-mode minibuffer-mode.lisp (set-url-mode form-fun-26) -
  SET-URL-MODE enabled.
<DEBUG> [19:06:04] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "e" NYXT::KEYCODE: 26 CHARACTER: #\e NYXT::KEYVAL-NAME: "e" 
<DEBUG> [19:06:04] nyxt input.lisp (dispatch-input-event) -
  Insert "e" in minibuffer
<DEBUG> [19:06:04] nyxt fuzzy.lisp (fuzzy-match) -
REDACTED
<DEBUG> [19:06:04] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "x" NYXT::KEYCODE: 53 CHARACTER: #\x NYXT::KEYVAL-NAME: "x" 
<DEBUG> [19:06:04] nyxt input.lisp (dispatch-input-event) -
  Insert "x" in minibuffer
<DEBUG> [19:06:04] nyxt fuzzy.lisp (fuzzy-match) -
REDACTED
<DEBUG> [19:06:04] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "a" NYXT::KEYCODE: 24 CHARACTER: #\a NYXT::KEYVAL-NAME: "a" 
<DEBUG> [19:06:04] nyxt input.lisp (dispatch-input-event) -
  Insert "a" in minibuffer
<DEBUG> [19:06:04] nyxt fuzzy.lisp (fuzzy-match) -
  ((https://example.com/  example domain 0.2916667)
    0.09049989))
<DEBUG> [19:06:05] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "m" NYXT::KEYCODE: 47 CHARACTER: #\m NYXT::KEYVAL-NAME: "m" 
<DEBUG> [19:06:05] nyxt input.lisp (dispatch-input-event) -
  Insert "m" in minibuffer
<DEBUG> [19:06:05] nyxt fuzzy.lisp (fuzzy-match) -
  ((https://example.com/  example domain 0.33333334))
<DEBUG> [19:06:05] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "p" NYXT::KEYCODE: 33 CHARACTER: #\p NYXT::KEYVAL-NAME: "p" 
<DEBUG> [19:06:05] nyxt input.lisp (dispatch-input-event) -
  Insert "p" in minibuffer
<DEBUG> [19:06:05] nyxt fuzzy.lisp (fuzzy-match) -
  ((https://example.com/  example domain 0.375))
<DEBUG> [19:06:05] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "l" NYXT::KEYCODE: 46 CHARACTER: #\l NYXT::KEYVAL-NAME: "l" 
<DEBUG> [19:06:05] nyxt input.lisp (dispatch-input-event) -
  Insert "l" in minibuffer
<DEBUG> [19:06:05] nyxt fuzzy.lisp (fuzzy-match) -
  ((https://example.com/  example domain 0.4166667))
<DEBUG> [19:06:05] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "e" NYXT::KEYCODE: 26 CHARACTER: #\e NYXT::KEYVAL-NAME: "e" 
<DEBUG> [19:06:05] nyxt input.lisp (dispatch-input-event) -
  Insert "e" in minibuffer
<DEBUG> [19:06:05] nyxt fuzzy.lisp (fuzzy-match) -
  ((https://example.com/  example domain 0.45833334))
<DEBUG> [19:06:05] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: NIL NYXT::KEYCODE: 50 CHARACTER: #\Nul
  NYXT::KEYVAL-NAME: "Shift_L" 
<DEBUG> [19:06:05] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "." NYXT::KEYCODE: 59 CHARACTER: #\.
  NYXT::KEYVAL-NAME: "period" NYXT::MODIFIERS: ("shift") 
<DEBUG> [19:06:05] nyxt input.lisp (dispatch-input-event) -
  Insert "." in minibuffer
<DEBUG> [19:06:05] nyxt fuzzy.lisp (fuzzy-match) -
  ((https://example.com/  example domain 0.5))
<DEBUG> [19:06:05] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "c" NYXT::KEYCODE: 54 CHARACTER: #\c NYXT::KEYVAL-NAME: "c" 
<DEBUG> [19:06:05] nyxt input.lisp (dispatch-input-event) -
  Insert "c" in minibuffer
<DEBUG> [19:06:06] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "o" NYXT::KEYCODE: 32 CHARACTER: #\o NYXT::KEYVAL-NAME: "o" 
<DEBUG> [19:06:06] nyxt input.lisp (dispatch-input-event) -
  Insert "o" in minibuffer
<DEBUG> [19:06:06] nyxt fuzzy.lisp (fuzzy-match) -
  ((https://example.com/  example domain 0.5416667))
<DEBUG> [19:06:06] nyxt fuzzy.lisp (fuzzy-match) -
  ((https://example.com/  example domain 0.5833334))
<DEBUG> [19:06:06] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "m" NYXT::KEYCODE: 47 CHARACTER: #\m NYXT::KEYVAL-NAME: "m" 
<DEBUG> [19:06:06] nyxt input.lisp (dispatch-input-event) -
  Insert "m" in minibuffer
<DEBUG> [19:06:06] nyxt fuzzy.lisp (fuzzy-match) -
  ((https://example.com/  example domain 0.625))
<DEBUG> [19:06:08] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "return" NYXT::KEYCODE: 36 CHARACTER: #\Return
  NYXT::KEYVAL-NAME: "Return" 
<DEBUG> [19:06:08] nyxt input.lisp (dispatch-input-event) -
  Found key binding "return" [#36] to RETURN-SELECTION
<DEBUG> [19:06:08] nyxt browser.lisp (request-resource) -
  Load URL from internal buffer in new buffer: https://example.com/
<DEBUG> [19:06:09] nyxt base-mode.lisp (base-mode form-fun-26) -
  BASE-MODE enabled.
<DEBUG> [19:06:09] nyxt/web-mode web-mode.lisp (web-mode form-fun-30) -
  WEB-MODE enabled.
<DEBUG> [19:06:09] nyxt/certificate-exception-mode certificate-exception-mode.lisp (certificate-exception-mode form-fun-36) -
  CERTIFICATE-EXCEPTION-MODE enabled.
<DEBUG> [19:06:09] nyxt/web-mode web-mode.lisp () -
  Notify URI https://example.com/ for buffer #<USER-WEB-BUFFER {100656AE93}> with load status UNLOADED
<DEBUG> [19:06:09] nyxt renderer-gtk.lisp (on-signal-decide-policy gtk-buffer) -
  Don't forward to #<USER-INTERNAL-BUFFER {1009665C93}>'s renderer (null request data).
<DEBUG> [19:06:09] nyxt browser.lisp (request-resource) -
  Forwarding https://example.com/ for buffer #<USER-WEB-BUFFER {100656AE93}>
<DEBUG> [19:06:09] nyxt renderer-gtk.lisp (on-signal-decide-policy gtk-buffer) -
  Forward to #<USER-WEB-BUFFER {100656AE93}>'s renderer (unchanged URL).
<DEBUG> [19:06:09] nyxt browser.lisp (request-resource) -
  Forwarding https://example.com/ for buffer #<USER-WEB-BUFFER {100656AE93}>
<DEBUG> [19:06:09] nyxt renderer-gtk.lisp (on-signal-decide-policy gtk-buffer) -
  Forward to #<USER-WEB-BUFFER {100656AE93}>'s renderer (unchanged URL).
 <INFO> [19:06:09] nyxt message.lisp (echo form-fun-6) -
  Loading "https://example.com/".
<DEBUG> [19:06:09] nyxt browser.lisp (request-resource) -
  Forwarding https://example.com/ for buffer #<USER-WEB-BUFFER {100656AE93}>
<DEBUG> [19:06:09] nyxt renderer-gtk.lisp (on-signal-decide-policy gtk-buffer) -
  Forward to #<USER-WEB-BUFFER {100656AE93}>'s renderer (unchanged URL).
<DEBUG> [19:06:09] nyxt/web-mode web-mode.lisp () -
  Notify URI https://example.com/ for buffer #<USER-WEB-BUFFER {100656AE93}> with load status LOADING
 <INFO> [19:06:09] nyxt message.lisp (echo form-fun-6) -
  Finished loading "https://example.com/".
^CUnhandled SB-SYS:INTERACTIVE-INTERRUPT in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                    {10006880D3}>:
  Interactive interrupt at #x7F34A3B40B9B.

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10006880D3}>
0: ("bogus stack frame")
1: ((FLET SB-UNIX::BODY :IN SB-THREAD::FUTEX-WAIT))
2: ((FLET "WITHOUT-INTERRUPTS-BODY-1" :IN SB-THREAD::%CONDITION-WAIT))
3: (SB-THREAD::%CONDITION-WAIT #<SB-THREAD:WAITQUEUE cl-cffi-gtk main thread {10082ECEA3}> #<SB-THREAD:MUTEX "cl-cffi-gtk main thread" (free)> NIL NIL NIL NIL NIL NIL)
4: ((FLET SB-THREAD::WITH-SYSTEM-MUTEX-THUNK :IN SB-THREAD::%DECREMENT-SEMAPHORE))
5: ((FLET "WITHOUT-INTERRUPTS-BODY-24" :IN SB-THREAD::CALL-WITH-SYSTEM-MUTEX/ALLOW-W
ITH-INTERRUPTS))
6: (SB-THREAD::CALL-WITH-SYSTEM-MUTEX/ALLOW-WITH-INTERRUPTS #<FUNCTION (FLET SB-THREAD::WITH-SYSTEM-MUTEX-THUNK :IN SB-THREAD::%DECREMENT-SEMAPHORE) {7F34A37B7ADB}> #<SB-THREAD:MUTEX "cl-cffi-gtk main thread" (free)>)
7: (SB-THREAD::%DECREMENT-SEMAPHORE #S(SB-THREAD:SEMAPHORE :%COUNT 0 :WAITCOUNT 1 :MUTEX #<SB-THREAD:MUTEX #1="cl-cffi-gtk main thread" (free)> :QUEUE #<SB-THREAD:WAITQUEUE #1# {10082ECEA3}>) 1 T NIL SB-THREAD:WAIT-ON-SEMAPHORE)
8: (SB-THREAD:WAIT-ON-SEMAPHORE #S(SB-THREAD:SEMAPHORE :%COUNT 0 :WAITCOUNT 1 :MUTEX #<SB-THREAD:MUTEX #1="cl-cffi-gtk main thread" (free)> :QUEUE #<SB-THREAD:WAITQUEUE #1# {10082ECEA3}>) :N 1 :TIMEOUT NIL :NOTIFICATION NIL)
9: (SB-THREAD:JOIN-THREAD #<SB-THREAD:THREAD "cl-cffi-gtk main thread" RUNNING {10082ECF13}> :DEFAULT NIL :TIMEOUT NIL)
10: (START :VERBOSE T :URLS NIL)
11: ((LAMBDA NIL :IN UIOP/IMAGE:RESTORE-IMAGE))
12: (UIOP/IMAGE:CALL-WITH-FATAL-CONDITION-HANDLER #<FUNCTION (LAMBDA NIL :IN UIOP/IMAGE:RESTORE-IMAGE) {10064C13CB}>)
13: ((FLET SB-UNIX::BODY :IN SB-IMPL::START-LISP))
14: ((FLET "WITHOUT-INTERRUPTS-BODY-1" :IN SB-IMPL::START-LISP))
15: (SB-IMPL::START-LISP)

unhandled condition in --disable-debugger mode, quitting
```

**Comment by aadcg** [comment]
Hi @edgar-vincent.  Thank you for sending us this bug report!

I can reproduce it. I'll take a look and come back with some insight.

**Comment by Ambrevar** [comment]
Fixed with 1212f4722a6c2936f603de5133f58cc3497db7bd.  Thanks for reporting!

**Comment by edgar-vincent** [comment]
Works fine! Thanks a lot!
